### ISSUE TITLE
RPC Bug when Calling RPC Function from Child Class

### ISSUE THREAD

**Issue by marcol1** [godotengine/godot#80597]
### Godot version

4.1.1.stable

### System information

Windows 11 10.0.22621 Build 22621

### Issue description

Currently experiencing an issue with RPCs and inheritance.

In the parent class, I have the following code:
@rpc("any_peer", "call_local") func Sync_addImpulse(swingImpulseString : String): if multiplayer.get_remote_sender_id() == 0: rpc("Sync_addImpulse", swingImpulseString) print("forwarding rpc") else: # some code print("some code")

When I call Sync_addImpulse in the child class, the function is called (e.g., "forwarding rpc" is printed). However, the RPC isn't received.

To fix, this, I simply echoed the call in the child as follows:
`

@rpc("call_local", "any_peer")
func Sync_addImpulse(impulse): super.Sync_addImpulse(impulse)

`

Strangely, it now works as expected. It seems like maybe the rpc tag isn't inherited by the child, but that's just a wild guess.

Let me know if you have any questions or if I can help at all.

### Steps to reproduce

- Create a parent & child class
- Create some method CallRPC() in the parent class, that echoes an RPC to itself (like the one I provided)
- Call the CallRPC() method from the child class. The parent class will run CallRPC(), but it won't receive the self-echoed RPC.

### Minimal reproduction project

N/A - quite trivial.

**Comment by marcol1** [comment]
ugh the code formatting isn't working, this is my first time commenting on GitHub, anyone know how to fix it? sorry!!

**Comment by AThousandShips** [comment]
You need triple backtics, I have fixed it for you

**Comment by marcol1** [comment]
@AThousandShips thanks for fixing it, appreciate that

**Comment by AThousandShips** [comment]
No problems! The GitHub syntax is confusing

**Comment by marcol1** [comment]
btw, even local calls don't work- hence I don't think name or nodepath are the issue

**Comment by marcol1** [comment]
strangely this was working before in my project. i'm looking into it more to see if I messed anything up.

**Comment by marcol1** [comment]
bumping because this is a pretty big deal.

**Comment by AThousandShips** [comment]
Please do not bump without adding any additional information, it is not forgotten

**Comment by marcol1** [comment]
![image](https://github.com/godotengine/godot/assets/39341403/c9244585-1e0e-416a-a102-6ca9e3339202)
Also, got this error.

It looks like the RPC tag isn't being inherited.

**Comment by marcol1** [comment]
update: having trouble reproducing in a small example. looking into this further, but it looks like there's definitely something strange and specific to my game happening here... going to keep trying to reproduce.

**Comment by marcol1** [comment]
@Dragoncraft89, @anvilfolk, forgive for the pings, just not sure where to direct this to. I'm thinking that this might be related to me making some changes in filenames and the engine not recompiling dependencies properly (?).

I've identified the commits that caused the bug and am currently trying to see if anything else works. Have had no luck reproducing in another project.

**Comment by anvilfolk** [comment]
Changing filenames is currently definitely associated with funky behavior! Godot caches information about which file defines each GDScript global class name. I've seen a few reports that renaming files (particularly `.gd` ones) results in that cache becoming invalid, and strange behavior results!

If you suspect this is the issue, maybe try renaming some files like how you might've done it originally? It would be nice to have a method to reproduce that kind of bad behavior so we can fix it also :)

**Comment by marcol1** [comment]
@anvilfolk interestingly, the filename change doesn't seem to be it, since I've identified (within 1-3 commits) where the regression was introduced in my project and there's no filename changes. I'll keep you posted. This is a nasty bug in any case, hoping it's just user error and not an engine thing.

**Comment by marcol1** [comment]
Wow, I found the source in my code:

```
var ReplicatedProperties : ReplicatedNodeProperties = ReplicatedNodeProperties.new(self)
```

declared outside of _ready, _init, etc. just part of the script body. Yet a similar setup isn't causing anything in my sample project. curious..

**Comment by marcol1** [comment]
update: planning on taking care of this & dedicating some hours in the next two weeks. pls do not close yet :)

**Comment by marcol1** [comment]
so i fixed it in my game, but have been unable to reproduce. i might get to it eventually.... if you _really_ want me to i can try lol.

It has something to do with including scene files in multiple places. But my game has so many moving parts I can't really elaborate further than that at this point.

I'll see if I can figure out how to make a repro project.

**Comment by anvilfolk** [comment]
Yeah, just holler if you eventually figure out where the issue is coming from! There's a tiny chance that #81201 fixes it, but I'm not certain. If you have a copy of the project from when it still wasn't working, you might be able to test whether it does by downloading the PR's build from [here](https://github.com/godotengine/godot/actions/runs/6148295286).

**Comment by marcol1** [comment]
hey @anvilfolk, ping me when that PR gets merged into the steam version maybe? If it's fixed then great! Otherwise I can try to reproduce, but this bug is _really_ abstract. I do want to help for sure, it's just a matter of having enough hours in the day :(

**Comment by anvilfolk** [comment]
This will only be in for 4.2, and it'll be a while until that is on Steam! Would you be able to backup your project and give it a whirl with whatever executable is appropriate for your OS from the bottom of this [page](https://github.com/godotengine/godot/actions/runs/6148295286)?

**Comment by anvilfolk** [comment]
If you're on Windows, you would use this link to download the Godot editor for windows: https://github.com/godotengine/godot/suites/16041453760/artifacts/917105316

**Comment by anvilfolk** [comment]
Actually, my bad, there was another regression that also got fixed, all of which deal with circular dependencies, so would you try [this one](https://github.com/godotengine/godot/suites/16085351045/artifacts/918308535) out?

**Comment by dev-mico** [comment]
Hey @anvilfolk, apologies for never responding here. This should be in the latest build, correct? If so, I'll add testing it to my TODO list so we can close this.

**Comment by anvilfolk** [comment]
@dev-mico totally cool :) Yes, it should be in the latest builds! Let us know if the issue still occurs! :)

**Comment by HolonProduction** [comment]
Closing due to lack of response, so I'm assuming this was resolved by the mentioned PR. If it still occurs, please let us know so that we can reopen the issue.
